modules/custom/d_commerce/modules/d_commerce_product/src/Entity/DrooplerProductVariation.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Drupal\d_commerce_product\Entity;
+
+use Drupal\commerce_product\Entity\ProductVariation as CommerceProductVariation;
+use Drupal\Component\Plugin\Exception\PluginException;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Logger\LoggerChannelTrait;
+
+/**
+ * Provides an override for commerce_product product variation entity class.
+ *
+ * @package Drupal\d_commerce_product\Entity
+ */
+class DrooplerProductVariation extends CommerceProductVariation {
+
+  use LoggerChannelTrait;
+
+  /**
+   * {@inheritDoc}
+   */
+  public function label(): string {
+    if ($this->shouldGenerateTitle()) {
+      $label = parent::label();
+    }
+    else {
+      $label = $this->createVariationLabel();
+    }
+
```

Comment:
A nie dało rady całość w jednej linii?:
`return $this->shouldGenerateTitle() ? parent::label() : $this->createVariationLabel();`